translations_parser/parser.py

```diff
@@ -0,0 +1,251 @@
+            logger.warning("Parsing logs from a newer version of Marian (> {MARIAN_MAJOR}.{MARIAN_MINOR})")
+
+        # Read Marian execution description on the next lines
+        desc = []
+        for headers, text in logs_iter:
+            if ("marian",) not in headers:
+                break
+            desc.append(text)
+
+        # Try to parse all following config lines as YAML
+        config_yaml = ""
+        while ("config",) in headers:
+            if "Model is being created" in text:
+                headers, text = next(logs_iter)
+                break
+            config_yaml += f"{text}\n"
+            headers, text = next(logs_iter)
+        try:
+            self.config = yaml.safe_load(config_yaml)
+        except Exception as e:
+            raise Exception(f"Invalid config section: {e}")
+
+        # Iterate until the end of file to find training or validation logs
+        while True:
+            if train := self.parse_training_log(headers, text):
+                self.training.append(train)
+            elif val := self.parse_validation_log(headers, text):
+                self.validation.append(val)
+            try:
+                headers, text = next(logs_iter)
+            except StopIteration:
+                break
+
+        count = sum(len(vals) for vals in self.indexed_logs.values())
+        logger.info(f"Successfully parsed {count} lines")
+        logger.info(f"Found {len(self.training)} training entries")
+        logger.info(f"Found {len(list(self.validation))} validation entries")
+        self.parsed = True
+
+    def parse(self):
+        """
+        Parse the log lines
+        A StopIteration can be raised if some required lines are never found
+        """
+        try:
+            self._parse()
+        except StopIteration:
+            raise ValueError("Logs file ended up unexpectedly")
+
+    @property
+    def validation(self):
+        """
+        Build validation entries from complete entries
+        as validation logs are displayed on multiple lines
+        """
+        for (epoch, up), parsed in self.validation_entries.items():
+            # Ensure required keys have been parsed
+            diff = set(("chrf", "ce_mean_words", "bleu_detok")) - set(parsed.keys())
+            if diff:
+                logger.warning(f"Missing keys for validation entry ep. {epoch} up. {up}: {diff}")
+                continue
+            yield ValidationEpoch(epoch=epoch, up=up, **parsed)
+
+    @property
+    def output(self):
+        if not self.parsed:
+            raise Exception("Please run the parser before reading the output")
+        return TrainingLog(
+            info=self.info,
+            configuration=self.config,
+            training=self.training,
+            validation=list(self.validation),
+            logs=self.indexed_logs,
+        )
+
+    def csv_export(self):
+        output = Path(__file__).parent.parent / "output"
```

Comment:
output path should be a variable, and available as CLI option (you can default to this folder)